tomviz/CentralWidget.cxx

```diff
@@ -257,14 +271,26 @@ CentralWidget::CentralWidget(QWidget* parentObject, Qt::WindowFlags wflags)
 
   this->EventLink->Connect(chart, vtkCommand::CursorChangedEvent, this,
                            SLOT(histogramClicked(vtkObject*)));
+
+  this->Worker->start();
+  this->HistogramGen->moveToThread(this->Worker);
+  this->connect(this->HistogramGen,
+     SIGNAL(histogramDone(vtkSmartPointer<vtkImageData>,vtkSmartPointer<vtkTable>)),
+     SLOT(histogramReady(vtkSmartPointer<vtkImageData>,vtkSmartPointer<vtkTable>)));
 }
 
 //-----------------------------------------------------------------------------
 CentralWidget::~CentralWidget()
 {
-  if (this->Worker && this->Worker->isRunning())
+  // disconnect all signals/slots
+  QObject::disconnect(this->HistogramGen, NULL, NULL, NULL);
+  QObject::connect(this->HistogramGen, SIGNAL(destroyed()),
+                   this->Worker, SLOT(quit()));
+  QMetaObject::invokeMethod(this->HistogramGen, "deleteLater");
```

Comment:
I'm not 100% sure you have to call deleteLater() on the thread that owns the object, but I'm playing it safe.